### PR TITLE
fix: guard EP activation checkpointing

### PR DESCRIPTION
**Pull request by WangXukang-cypher**
## Summary

Add early validation for EP training configurations that combine `ep_size > 1` with full activation checkpointing under HybridEP/DeepEP dispatch. This fails fast with a clear error instead of surfacing a backward-time checkpoint shape mismatch.

Update affected SFT/RL example scripts to default gradient checkpointing to `none` for EP configurations, while documenting that full checkpointing can still be used with `MOLT_MOE_DISPATCHER=torch`.

## Test Plan

- `python -m py_compile molt\cli\common_args.py molt\cli\train_rl_ray.py molt\cli\train_sft.py tests\unit\test_ep_ac_validation.py`
- Not run; reason: local environment is missing `pytest`.

### CODE REVIEW

**Comment by hijkzzz** [comment]
please use our docker image (cuda13) with deepep/hybridep~
also cc @huiying

**Comment by WangXukang-cypher** [comment]
Thanks for the guidance. I will re-test this with the project CUDA 13 Docker image and DeepEP/HybridEP enabled, then update the PR based on the result.
